app/assets/javascripts/components/expander.js

```diff
@@ -68,6 +68,27 @@ window.GOVUK.Modules = window.GOVUK.Modules || {}; // if this ; is omitted, none
     $button.setAttribute('type', 'button')
     $button.setAttribute('aria-expanded', expanded)
     $button.setAttribute('aria-controls', this.$content.getAttribute('id'))
+
+    // GA4 Accordion tracking. Relies on the ga4-finder-tracker setting the index first, so we wrap this in DOMContentLoaded.
+    window.addEventListener('DOMContentLoaded', function () {
```

Comment:
Yep, we have to make sure that `window.GOVUK.analyticsGa4.Ga4FinderTracker.setFilterIndexes` has run first, and this was the easiest way to ensure that.

Reply:
Although like the cookie code, you could use a variable to avoid the timing problem.

Reply:
Thanks @andysellick - I've added this now. Not sure what you mean about using a variable for a timing problem though!

Reply:
I think what I meant was we know that DOMContentLoaded fires after all the JS has loaded, so it's a safe bet to rely on it to do this. But are we sure that with this approach that the custom event will fire after the event listeners that depend on it have been created? We might be, I'm not certain (sorry, sounds like I'm criticising my own suggestion here, confusing). That would look like this:

- event listeners get set up for `ga4-filter-indexes-added`
- event is triggered, all works

If that happened the other way around, we'd need something slightly more complex

- `ga4-filter-indexes-added` event is triggered
- at the same time, set something in the `window.GOVUK` object to say that it has been triggered, e.g. `window.GOVUK.analyticsGa4.vars.ga4-filter-indexes-added = true`
- code with event listeners first checks for the above var, if it finds it then execute immediately, otherwise create event listeners

I think that would work. However, I might be overcomplicating this and it's probably quite easy to look at the order the scripts load in `finder-frontend` to have assurance that the first scenario (what you've got now) is fine. So probably no changes needed, I just wanted to explain what I was thinking. Your change is good, by the way 😁

Reply:
@andysellick Oh - I understand now. I think the order that the JS loads should make what we've currently done OK, this is what it looks like:

```JavaScript
//= require_tree ./analytics-ga4
//= require live_search
```

https://github.com/alphagov/finder-frontend/blob/3357f3693dc6e311bc2465662f166ac305a1dbce/app/assets/javascripts/application.js#L22

Reply:
Awesome, was hoping that would be the case 👍